### ISSUE TITLE
Changes to trivy CRDs by the operator happen rather frequently causing other applications to be impacted.

### ISSUE THREAD

**Issue by andy-v-h** [aquasecurity/trivy-operator#315]
**What steps did you take and what happened:**

The `v0.1.3` was released into our environment today, and shortly thereafter argocd started burning a lot of CPU. To the point, it was pressing up towards pod evictions. After investigating, I found that argocd will [report what triggers](https://github.com/argoproj/argo-cd/pull/8192/files) the `application-controller` to do reconciliation in debug mode. Minutes after restarting my pods in debug reporting, the most common resource to be updated that triggers a reconciliation are aqua security CRDs

```bash
k logs po/argocd-application-controller-2 | rg "Refreshing app" | awk -F "type" '{print $2}' | awk -F '"' '{print $1}' | sort | uniq -c                                                                           
1236  aquasecurity.github.io/v1alpha1/ClusterRbacAssessmentReport
6478  aquasecurity.github.io/v1alpha1/ConfigAuditReport
2316  aquasecurity.github.io/v1alpha1/RbacAssessmentReport
   1  autoscaling/v1/HorizontalPodAutoscaler
   6  ceph.rook.io/v1/CephCluster
   7  ceph.rook.io/v1/CephObjectStore
2090  v1/ConfigMap
```

This impact is very dependent on the use of argocd in my environment, however, the impact is incredible. And I'm certain other trivy users will also be argocd users.
![image](https://user-images.githubusercontent.com/5001148/178753141-69ca5ecf-5fa6-4134-b064-948595d607d2.png)

And once I removed `trivy-operator` from my clusters, the CPU patterns returned to pre-v0.1.3 levels 
![image](https://user-images.githubusercontent.com/5001148/178753683-a67cf833-fd53-4fbb-8445-19b28acaf313.png)

**What did you expect to happen:**

I wouldn't expect an update to trivy to have such a broad impact on other tools.

**Anything else you would like to add:**

There may be additional configuration options on argocd may provide a workaround. However, I wanted to file this report as I am not aware of how the v0.1.3 update may impact tools other than argocd. A resolution here maybe documentation or argocd users to deploy trivy-operator along with argocd and avoid this high CPU problem.

**Environment:**

- Trivy-Operator version (use `trivy-operator version`): v0.1.3
- Kubernetes version (use `kubectl version`): 1.21-1.23
- OS (macOS 10.15, Windows 10, Ubuntu 19.10 etc): flatcar v3132

**Comment by chen-keinan** [comment]
@andy-v-h its not trivy 0.1.3 in specific, what you see is the misconfiguration and rbac assessment checks running very quickly.
The cluster peak is only on startup when the operator hit misconfiguration and rbac resources. afterwards its only delta. maybe it is require to scale up `trivy-operator` for that peak and then scale down once completed.
is should be very fast.

**Comment by chen-keinan** [comment]
@andy-v-h was it a peak for short period of time or for a long time , its not quite clear from the graph above ?

**Comment by andy-v-h** [comment]
it was for ~7hours

**Comment by chen-keinan** [comment]
thanks for clarification , now it much clear.

**Comment by chen-keinan** [comment]
@andy-v-h the fix for  #357 the overload , we will do a patch release ASAP

**Comment by JAORMX** [comment]
By the end of the week I'll work with @andy-v-h to re-deploy trivy-operator in some test clusters, see if the PR addresses the issue alongside the argoCD ignore statements.

**Comment by chen-keinan** [comment]
close this issue as fix was provided by #357 , if still and issue can be opened again
